src/test/java/org/springframework/retry/policy/MockNeverRetryPolicy.java

```diff
@@ -0,0 +1,9 @@
+package org.springframework.retry.policy;
+
+import org.springframework.retry.RetryContext;
+
+public class MockNeverRetryPolicy extends NeverRetryPolicy {
```

Comment:
I'm not sure why we need this class as top-level one: probably really better to have it as inner one, like that `MockRetryCallback` in the `CircuitBreakerRetryTemplateTests`. But still let's reconsider if we just can deal with plain `NeverRetryPolicy`!

Reply:
Hi @artembilan, I did the mock for this class because I wanted to show you fix the bug (NPE), then in this case every executions in the test will enter in that point. no matter what class you use, the important is show the fix.  if you want, can you run the test without my fix for you understand the problem and the solution (and the test).

Reply:
likewise I will change the class with your suggestion